src/config.ts

```diff
@@ -240,3 +241,17 @@ export const manageFlowEnableUserList = withDefault(
       `Invalid Manage Flow enable user list configured: ${process.env.MANAGE_FLOW_ENABLE_USER_LIST}`
     );
   });
+
+/**
+ * Lock the creation of a new APIM user, when resolve SelfCareIdentity.
+ */
+export const lockSelfcareCreateNewApimUser = withDefault(
+  BooleanFromString,
+  ("false" as unknown) as boolean
```

Comment:
Unfortunately not, I tried locally by logging the value in the 2 use cases _(1.no env key, 2.with env key)_ and as proposed by you it doesn't work because it is necessary to parse the string first _(throw the Error)_.
If I use `t.boolean` instead of `BooleanFromString` I have the reverse problem: the default `false` works but it fails if I specify a value from environment _(which is seen as a string)_.

We have already addressed this aspect in the **io-services-cms** repo and this is the solution we adopt in this type of _"withDefault"_ scenario where the key value is different from string.

Reply:
Ok, I got it. It's a known bug of `withDefault` (I remember now). That happens because it doesn't consider codecs with asymmetric types (like this boolean/string). The bad part is that you are casting "false" (a string) into a boolean 🙃 

But we can't do otherwise unfortunately